datalad/downloaders/tests/test_http.py

```diff
@@ -311,10 +311,16 @@ def test_download_ftp():
         import requests_ftp
     except ImportError:
         raise SkipTest("need requests_ftp")  # TODO - make it not ad-hoc
-    yield check_download_external_url, \
-          "ftp://ftp.gnu.org/README", \
-          None, \
-          "This is ftp.gnu.org"
+    # Started to throw 504 when on travis
+    try:
+        yield check_download_external_url, \
```

Comment:
I'd guess that the tests are still failing because you're trying to catch the exception around the `yield` rather than the actual invocation.  Given that nothing else is yielded in this test, can you call `check_download_external_url` directly instead?